src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,239 @@
+
+/**
+ * Builds the forwards continuous velocity profile for the given parameters.
+ * NOTE: The velocity profile returned by this function is NOT reverse continuous and does
+ * not guarantee a feasible path to follow.
+ *
+ * NOTE: The velocity_profile wil contain the same number of real elements as the input
+ * max_allowable_speed[] array
+ *
+ * @pre The velocity_profile array is pre-allocated to contain up to
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS elements
+ *
+ * @param num_segments [in] The number of segments(elements) in the
+ * max_allowable_speed_profile input array
+ * @param velocity_profile [out] The array that the forwards continuous array will be
+ * copied into
+ * @param max_allowable_speed_profile [in] The pre-allocated array that will contain all
+ * of the maximum allowable velocity values for each point on th curve
+ * @param arc_segment_length [in] The lenmgth of each arc length segment in meters
+ * @param max_allowable_acceleration [in] The max allowable acceleration at any point in
+ * m/s^2
+ * @param max_allowable_speed [IN] The max allowable speed at any point in m/s
+ *
+ * */
+void app_trajectory_planner_generate_forwards_continuous_velocity_profile(
+    unsigned int num_segments, float* velocity_profile,
+    float* max_allowable_speed_profile, const float arc_segment_length,
+    const float max_allowable_acceleration, const float max_allowable_speed);
+
+/**
+ * Modifies a forwards continuous velocity profile to also be backwards continuous based
+ * on the input parameters.
+ *
+ * @pre The input velocity_profile is required to be FORWARDS CONTINUOUS before it is used
+ * as an input to this function
+ *
+ * @param num_segments [in] The number of segments(elements) in the velocity_profile array
+ * @param forwards_continuous_velocity_profile The forwards continuous velocity profile to
+ * ebe modified in-place into a profile that is also backwards continuous
+ * @param arc_segment_length [in] The arc segment length of each segment in the path the
+ * profile is being generated for in meters
+ * @param max_allowable_acceleration [in] The max allowable acceleration at any point on
+ * the velocity profile
+ */
+void app_trajectory_planner_generate_backwards_continuous_velocity_profile(
+    unsigned int num_segments, float* forwards_continuous_velocity_profile,
+    const float arc_segment_length, const float max_allowable_acceleration);
+
+/**
+ *  Adds a time profile to an existing position profile for the given input parameters
+ *
+ *  @pre traj_elements must be pre-allocated and contain all of the position data of the
+ * trajectory
+ *
+ * @param traj_elements The trajectory element array that will be modified to contain the
+ * time profile
+ * @param num_segments [in/ The number of segments(elements) in traj_elements and
+ * velocity_profile
+ * @param arc_segment_length  [in] The arc length of each path segment
+ * @param velocity_profile  The forwards and backwards continuous velocity profile of the
```

Comment:
Seems like the forwards and backwards continuity is a pre-condition?

Reply:
I've added the @pre for backwards and forwards continuity. Technically the function works without those, but the actual trajectory may not be functional

Reply:
Ah gotcha. If it's not a precondition for the algorithm that results in undefined behavior if broken, I'm inclined to not list it as a precondition.

@garethellis0 for his opinion

Reply:
Removed